lighthouse-core/gather/gatherers/form-elements.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license Copyright 2019 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+
+/* eslint-env browser, node */
+
+/**
+ *  @param {HTMLElement}
+ *  @return {String}
+ */
+/* istanbul ignore next */
+function getParentForm(node) {
```

Comment:
various input elements have a `.form` attribute that points to their containing/associated form

https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fae-form

![image](https://user-images.githubusercontent.com/39191/86653407-051ee100-bf9a-11ea-8850-86a32e80e8c7.png)

Reply:
wow I wish I knew this earlier that would have saved a lot of time. let me integreate this into the gatherer